### ISSUE TITLE
NumPy.org site map - tracking issue

### ISSUE THREAD

**Issue by rgommers** [numpy/numpy.org#43]

**Comment by InessaPawson** [comment]
The latest version of the site map is stored here: https://app.flowmapp.com/share/e1b61759a5f95f43d8907baedf42875d/sitemap/

**Comment by Shekharrajak** [comment]
> The latest version of the site map is stored here: https://app.flowmapp.com/share/e1b61759a5f95f43d8907baedf42875d/sitemap/

This is looking nice. It would be great if you can write few points about the page and may be the existing working URL of the corresponding page in the labels. e.g. I didn't get the `Array Computing ` page - means what will be the content there (I can't find the corresponding page in current site)

**Comment by InessaPawson** [comment]
Thank you for your feedback, @Shekharrajak! Yes, I'm currently working on including more info + content for each page. As far as linking to the existing working URLs most of the proposed is brand new to numpy.org. 
“Array Computing” page is proposed in NEP 28 (numpy.org/neps/nep-0028-website-redesign.html). @rgommers would be the best person to answer any questions about it. I haven’t got to it just yet.:)

**Comment by rgommers** [comment]
That site map looks cool!

There some content I drafted for `About Us`, `Community`, `Install` and `Learning` at https://github.com/rgommers/numpy.org/tree/newsite-content/www

> “Array Computing” page is proposed in NEP 28 (numpy.org/neps/nep-0028-website-redesign.html). @rgommers would be the best person to answer any questions about it.

This would have some general content about what array computing (i.e. computing with n-dimensional arrays, the core of what NumPy is about) and how NumPy fits into the whole ecosystem here and what related projects are. Have a look at "Integrates with existing projects" and "Powered by Dask" on https://dask.org/ to get an idea of the type of content I'm thinking about.

**Comment by rgommers** [comment]
A few thoughts on the site map:

- Looks quite complete, I'm not missing anything important.
- `Community Survey`: running the survey is a time-limited thing, but publishing the results on this page sounds like a really nice idea
- `Case Studies`: would put that on the front page, or linked from front page via an image/summary
- `Annual Reports`: the BIDS grant reports can be a good start for that. Also links to the NumFOCUS annual reports. I'd rename it though, to avoid giving the impression or making the commitment that we will write an annual report (can't really do that right now)
- `Tools` and `Libraries and Extensions` could be merged, not sure what would go where otherwise.

**Comment by datapythonista** [comment]
We discussed with @rgommers that it'd be good for users to have the navigation somehow standardized among projects (mainly for users common to different projects).

You structure looks quite good to me, I like some of the ideas and I think we can adopt them.

But my feeling is that 8 items in the top navigation are too many. In pandas currently we have 4, and I don't think we should have many more. You can see them here: https://dev.pandas.io/ I see Dask having 6, and as a user I feel it starts making things a bit too complex. Do you have a prototype to see that structure implemented already?

**Comment by rgommers** [comment]
> Do you have a prototype to see that structure implemented already?

Not yet; the outlines are starting to become visible in the `newsite` branch in this repo.

**Comment by rgommers** [comment]
> But my feeling is that 8 items in the top navigation are too many. In pandas currently we have 4, and I don't think we should have many more. You can see them here: https://dev.pandas.io/ I see Dask having 6, and as a user I feel it starts making things a bit too complex.

Not so sure, I quite like the Dask site. It's still on the minimal side I think. https://jupyter.org/ and https://julialang.org/ have 8 in the menu bar, https://quantecon.org/ has 6, python.org has two menu bars with 6 entries in the top one and 7 in the bottom one.

So I think 6 is the absolute minimum we can get away with; anything in the 6-8 range is fine.

https://dev.pandas.io/ has 4 in the top bar, but then another set of navigation links under the install button, and more in the side bar. It looks quite a bit busier than the Dask site.

**Comment by datapythonista** [comment]
I think in pandas we like things a bit more minimalist, you can see it in the API. ;)

To focus a bit more the conversation, since the size of the nav bar seem to be a question of preference, I see that those should be non-controversial:

- Install
- Documentation
- Community
- Donate

Some of the differences are:

- About us at the top level: We don't have that much there yet, but I guess it can make sense for us. Just the blog, I think it belongs to the community in our case, since it'll be an aggregator of other blogs.
- Contribute: Do you have it at the top level to make it more visible, or do you think doesn't belong to Community (where we have it)?
- Learning: If I get it right, besides the tutorials, you'd have there a page with links to stackoverflow and discourse, and another for what we call ecosystem. I'd personally look for those two in community. Is any other project using this structure? I see Julia has a learning section, but I think only for tutorials
- I think if we wanted an equivalent of "Array computing" will be in the same ecosystem page. But I guess it's more important for you, and I think it's fine that we differ on that.

**Comment by rgommers** [comment]
>     * Contribute: Do you have it at the top level to make it more visible, or do you think doesn't belong to Community (where we have it)?

Yes, I think this could be folded into Community, that would make sense. Visibility was a thought, but I didn't think all that hard about it.



>     * Learning: If I get it right, besides the tutorials, you'd have there a page with links to stackoverflow and discourse, and another for what we call ecosystem. I'd personally look for those two in community. Is any other project using this structure? I see Julia has a learning section, but I think only for tutorials

I indeed like the way Julia does it. Tutorials, videos, books, courses, cheat sheets. There's enough scope there that I wouldn't just put it under Documentation.

Tools + Libraries/extensions --> Ecosystem does make sense to me. It's not clear that it needs to be under Learning, could also be under Community or, maybe best, on the main page like on https://julialang.org/.

>     * About us at the top level: We don't have that much there yet, but I guess it can make sense for us. Just the blog, I think it belongs to the community in our case, since it'll be an aggregator of other blogs.

Jupyter has Donate under About Us. I like the way Jupyter does it. And now that I checked, that's how I wrote the content already: https://github.com/rgommers/numpy.org/blob/newsite-content/www/about.md

**Comment by InessaPawson** [comment]
> A few thoughts on the site map:
> 
> * Looks quite complete, I'm not missing anything important.
> * `Community Survey`: running the survey is a time-limited thing, but publishing the results on this page sounds like a really nice idea
Yes, that is exactly what I had in mind for the Community Survey page.

> * `Case Studies`: would put that on the front page, or linked from front page via an image/summary
Makes a lot of sense! 

> * `Annual Reports`: the BIDS grant reports can be a good start for that. Also links to the NumFOCUS annual reports. I'd rename it though, to avoid giving the impression or making the commitment that we will write an annual report (can't really do that right now)
Instead of a separate page, let’s have sections on the Donate page linking to NumFocus and BIDS annual reports. @rgommers, what do you think?

> * `Tools` and `Libraries and Extensions` could be merged, not sure what would go where otherwise.
Done!

**Comment by datapythonista** [comment]
Thanks @rgommers, good feedback.

I think something like this makes sense to me (and to you for what you say):

- Install
- Learning (probably a single page)
  - Tutorials
  - Videos
  - Books
  - Courses
  - Cheat sheets
- Documentation
  - User guide
  - API reference
  - Release notes
- Community
  - Blog
  - Ask a question (StackOverflow)
  - Discuss (Discourse)
  - Contributing
  - Code of conduct
  - Community survey
- About us
  - Our history
  - Ecosystem
  - Roadmap
  - Team
  - Sponsors
  - Citing pandas
  - Donate

You'd have the Array computing in the top level too. And there are minor things like whether donate is in the top level or in about us (I think some extra money from having in the top level would be nice). Or the roadmap (you've got it in community, but I guess may make more sense in About us).

Is there anything that you would change?

**Comment by rgommers** [comment]
> > `Annual Reports`: the BIDS grant reports can be a good start for that. Also links to the NumFOCUS annual reports. I'd rename it though, to avoid giving the impression or making the commitment that we will write an annual report (can't really do that right now)
> 
> Instead of a separate page, let’s have sections on the Donate page linking to NumFocus and BIDS annual reports. @rgommers, what do you think?

Makes sense, let's do that.


> You'd have the Array computing in the top level too. And there are minor things like whether donate is in the top level or in about us (I think some extra money from having in the top level would be nice). Or the roadmap (you've got it in community, but I guess may make more sense in About us).
> 
> Is there anything that you would change?

That looks pretty good. Only minor tweaks perhaps:

- Documentation will have some more things for us, like NEPs, f2py guide, and links to older versions of the docs
- Ecosystem: on front page rather than About Us
- Roadmap: not sure, perhaps linked from front page. About Us doesn't seem like the most obvious place.

**Comment by datapythonista** [comment]
Makes sense, we'll also have to link to older versions of the documentation. I added links to the home page, but I think having it under documentation also makes sense. I updated the tree with the sections specific to NumPy.

For the ecosystem and the roadmap, when you say having them in the front page, you mean in the home, like in the navigation we've got in the middle of dev.pandas.io? Not sure if About us is the best place, but I guess they should be in the navigation bar too. Don't you think so?

**Comment by rgommers** [comment]
> For the ecosystem and the roadmap, when you say having them in the front page, you mean in the home, like in the navigation we've got in the middle of dev.pandas.io? Not sure if About us is the best place, but I guess they should be in the navigation bar too. Don't you think so?

No not navigating to it from anywhere, the actual content present on `numpy.org`. See how Ecosystem is done on https://julialang.org/. 

I think it's a bit of a waste that there are only links on dev.pandas.io itself, why not put something interesting there?

**Comment by datapythonista** [comment]
Ah, I see, thanks for clarifying. Our new home is far from finished, we're still discussing what to have there (https://github.com/pandas-dev/pandas/issues/28168).

I added having the ecosystem too, I think it's a good idea. But our ecosystem is huge, we can have the basics in the home, but we'll need to have a separate page too. And I think would be nice to have it in the navigation.

For the roadmap, I see it complements well "Our history" in About us. Seems  like Our history is the past of the project, and the roadmap is the future of the project itself, not so much a community thing (in that sense). We have a link from donate, and having a link in the home sounds good to. But as with the ecosystem, I think it should also be in the navigation.

**Comment by datapythonista** [comment]
Did a quick test with our website, and I think the size of the navbar and the dropdowns is great with the current proposal: https://datapythonista.github.io/pandas-web/ (most of the links will be broken, but gives an idea of the navigation).

**Comment by InessaPawson** [comment]
> * Contribute: Do you have it at the top level to make it more visible, or do you think doesn't belong to Community (where we have it)?

I love, preach, and practice intuitive, clean and minimalistic UX. While designing the top navigation bar, I’ve carefully considered the needs of all NumPy stakeholders, including the project itself. Our current most pressing needs are contributors and funding. By placing the “Contribute” button in the top navigation bar we will plant an idea in the head of those who haven’t considered contributing before, and we will make it easy to sign up for those who are ready. Just like by placing the “Donate” button will indicate to the new community members that we are a non-profit with all the opportunities and constraints that come with this kind of business model. Hopefully we will get some donations as well.

> * Learning: If I get it right, besides the tutorials, you'd have there a page with links to stackoverflow and discourse, and another for what we call ecosystem. I'd personally look for those two in community. Is any other project using this structure? I see Julia has a learning section, but I think only for tutorials

Students and educators are a large and probably the fastest growing part of the NumPy community. Placing educational materials under “Community” is not intuitive enough for this group, especially if English is their second language, and they are new to Data Science and/or open source. 

> * Just the blog, I think it belongs to the community in our case, since it'll be an aggregator of other blogs.

I was considering to place “Blog/Newsletter” as a child page of “Community”. (@rgommers  I’m leaning towards naming the button as “Newsletter”. ) Here is why I’ve decided that it belongs to the “About Us” section: we are planning to spotlight our contributors in the newsletter, which would be complementing “Our team” page in the same menu.

>* For the roadmap, I see it complements well "Our history" in About us. 

Agreed! There is an existential balance in having “History” and “Roadmap” in the same navigation bar. :)

> * I think if we wanted an equivalent of "Array computing" will be in the same ecosystem page. But I guess it's more important for you, and I think it's fine that we differ on that.

We might make “Array Computing” a child page or section of another web page. We need to finish writing up our content to make the final decision.

**Comment by InessaPawson** [comment]
>*Roadmap: not sure, perhaps linked from front page. About Us doesn't seem like the most obvious place.

@rgommers I think we should do both.

**Comment by InessaPawson** [comment]
> Tools + Libraries/extensions --> Ecosystem does make sense to me. It's not clear that it needs to be under Learning, could also be under Community or, maybe best, on the main page like on https://julialang.org/.

My vote goes to the third option - placing it on the main page. It should be a part of the intro to NumPy.

**Comment by InessaPawson** [comment]
>Install
Learning
Documentation

@rgommers @datapythonista  Is this order a go? Frankly, I quite liked having "Learning" and "Community" next to each other.

**Comment by rgommers** [comment]
> I love, preach, and practice intuitive, clean and minimalistic UX. While designing the top navigation bar, I’ve carefully considered the needs of all NumPy stakeholders, including the project itself. Our current most pressing needs are contributors and funding. By placing the “Contribute” button in the top navigation bar we will plant an idea in the head of those who haven’t considered contributing before, and we will make it easy to sign up for those who are ready.

This is a very good point. We do _really_ need to gain more contributors to various areas of the project. Okay, you've convinced me:)

Agreed on the Roadmap in both places and importance of top-level Learning too.

> Just like by placing the “Donate” button will indicate to the new community members that we are a non-profit with all the opportunities and constraints that come with this kind of business model. Hopefully we will get some donations as well.

Donations are unlikely to be a major part of the kind of revenue stream we need, and I don't think it gives the right signal to larger potential funders. So if it's top-level, it should be called Support or Funding or some such phrase.

> @rgommers @datapythonista Is this order a go? Frankly, I quite liked having "Learning" and "Community" next to each other.

I didn't think much about order. Learning and Community next to each other sounds good to me.

**Comment by datapythonista** [comment]
Good feedback, thank you. I made the donation pages a starting point for anyone supporting the project, linking to contribue to. I think pandas has enough contributors (like first timers), and individual donations are also not very significant. I think what makes sense to me is to have contributing in the top level, and make the home page of contributors similar to what is the donations page now (do you want to contribute time? money? ...)

Didn't think about the order much either. But I've been sharing the draft in dev.pandas.io and experimenting with it myself, and it doesn't feel very natural, I think what makes sense in our case is:

- About us
  - Our history
  - Ecosystem
  - Roadmap
  - Team
  - Sponsors
  - Citing pandas
  - Donate
- Getting started (what was Install + Learning before)
  - Install
  - Tutorials
  - Videos
  - Books
  - Courses
  - Cheat sheets
- Documentation
  - User guide
  - API reference
  - Release notes
  - Older versions
  - ... (NEPs, f2py guide for NumPy...)
- Community
  - Blog
  - Ask a question (StackOverflow)
  - Discuss (Discourse)
  - Code of conduct
  - Community survey
- Contributing
- ... (Array computing for NumPy - NBViewer, JupyterHub, Widgets for Jupyter,...)

For what you say, and given that NumPy is much more academic than pandas, I guess you prefer the separate learning section in the top level. But to me it doesn't feel natural, since our install is quite minimal, and the feedback about the learning section wasn't positive.

You can give it a try at dev.pandas.io. I'll probably implement the new structure this evening (Europe).

**Comment by InessaPawson** [comment]
> Donations are unlikely to be a major part of the kind of revenue stream we need, and I don't think it gives the right signal to larger potential funders. So if it's top-level, it should be called Support or Funding or some such phrase.

Let's fold it under "Contribute" then.

**Comment by amueller** [comment]
I haven't followed the discussion in detail, but just some thoughts.
I'm by no means a UI or UX designer, but I think that user stories are a good design tool. Have you considered the different user stories?

@InessaPawson also brought up that the menu and landing page are places to create a call for action, which might be another important component. I think I would be more concerned with usability, though.

sklearn just changed it's navigation somewhat, but this is not final yet:
https://scikit-learn.org/dev/

Personally I think that very few people are interested in the "about us" part.
Re donate: People I talk to are regularly shocked by the lack of support for core projects like numpy, but I don't think having a "donate" button, no matter how visible, will change that. Also, you shouldn't expect any significant donations to come from that (sklearn has had a button for many years, in various places. I think NumFocus still has the funds from what came in through that because it never was an amount big enough for us to think about transferring it; probably less than $1k over 7 years or so? I would need to check).

I like having "ask a question" in the menu. We have that only in the FAQ. I think that's probably a very common use-case of people coming to the website and having this more prominent makes sense. At least for sklearn none of the other points under "community" seem very relevant, though.

**Comment by datapythonista** [comment]
I think we all agree on the donate. And how I see it is perfectly fine for users if some projects don't have some sections. I think the value is that we don't have the Roadmap in about us in one project, and we call it "Future plans" and have it in the community in another. I think that's unnecessary and makes sense to coordinate. But I assume some differences make sense, since the projects are different.

I already got a draft for pandas following more or less what we discussed. Didn't make sense to me to have Install + Learning, and I have a Getting started with both instead. But for the rest I think it's what we discussed.

https://datapythonista.github.io/pandas-web/

**Comment by rgommers** [comment]
> I'm by no means a UI or UX designer, but I think that user stories are a good design tool. Have you considered the different user stories?

There's a start in gh-42, and we asked on the mailing list for people to contribute more stories. So far not much response though.

> Re donate: People I talk to are regularly shocked by the lack of support for core projects like numpy, but I don't think having a "donate" button, no matter how visible, will change that.

It requires asking and having a concrete goal and ways of spending the money. Bokeh just raised over $2,000 in one month (their goal was $1,000) by asking people to support the project to cover web hosting etc.

That said, I agree that it's still minor compared to other potential funding sources. We should prioritize those.

**Comment by InessaPawson** [comment]
> I'm by no means a UI or UX designer, but I think that user stories are a good design tool. Have you considered the different user stories?

We could certainly write 20-25 user stories ourselves (adequate for this size of a project). However, it made sense for this target audience to send out an invitation for collaboration first.

> Personally I think that very few people are interested in the "about us" part.

@amueller Please, never underestimate human curiosity!:) I happened to have 10+ experience in web development and digital marketing. It is very common for About Us to be the second most visited page on a website. It rings true across a wide range of industries, from luxury travel to biotech and higher education.

**Comment by InessaPawson** [comment]
We will be updating the sitemap as more pages are added. @rgommers , should we keep this issue open?

**Comment by rgommers** [comment]
Let's leave it open for now; makes it easy to find the sitemap, and I think we should revisit this after launching the new site, there's some useful discussion about site structure

**Comment by InessaPawson** [comment]
Thank you to everyone who contributed to this discussion! The website has evolved quite a bit since this issue was first opened. Let's close it now. If there are specific sitemap improvements that need to be made, we can open new, more focused issues.
